R/test-module.R

```diff
@@ -0,0 +1,117 @@
+
+
+#' Test a shiny module
+#' @param module The module under test
+#' @param expr Test code containing expectations. The test expression will run
+#'   in the module's environment, meaning that the module's parameters (e.g.
+#'   `input`, `output`, and `session`) will be available along with any other
+#'   values created inside of the module.
+#' @param args A list of arguments to pass into the module beyond `input`,
+#'   `output`, and `session`.
+#' @param ... Additional named arguments to be passed on to the module function.
+#' @include mock-session.R
+#' @export
+testModule <- function(module, expr, args, ...) {
+  expr <- substitute(expr)
+  .testModule(module, expr, args, ...)
+}
+
+#' @noRd
+#' @importFrom withr with_options
+.testModule <- function(module, expr, args, ...) {
+  # Capture the environment from the module
+  # Inserts `session$env <- environment()` at the top of the function
+  fn_body <- body(module)
+  fn_body[seq(3, length(fn_body)+1)] <- fn_body[seq(2, length(fn_body))]
+  fn_body[[2]] <- quote(session$env <- environment())
+  body(module) <- fn_body
+
+  # Substitute expr for later evaluation
+  if (!is.call(expr)){
+    expr <- substitute(expr)
```

Comment:
Isn't this already handled by `testModule`?

Reply:
Good catch! Dead code that I forgot to pull out.